codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ManifestWriter.java

```diff
@@ -0,0 +1,141 @@
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.node.StringNode;
+import software.amazon.smithy.model.traits.UnstableTrait;
+
+/**
+ * Generates a manifest description of the generated code, minimum go version, and minimum dependencies required.
+ */
+public final class ManifestWriter {
+    private static final String GENERATED_JSON = "generated.json";
+
+    private ManifestWriter() {
+    }
+
+    /**
+     * Write the manifest description of the generated code.
+     *
+     * @param settings     the go settings
+     * @param model        the smithy model
+     * @param fileManifest the file manifest
+     * @param dependencies the list of symbol dependencies
+     */
+    public static void writeManifest(
+            GoSettings settings,
+            Model model,
+            FileManifest fileManifest,
+            List<SymbolDependency> dependencies
+    ) {
+        Path manifestFile = fileManifest.getBaseDir().resolve(GENERATED_JSON);
+
+        if (Files.exists(manifestFile)) {
+            try {
+                Files.delete(manifestFile);
+            } catch (IOException e) {
+                throw new CodegenException("Failed to delete existing " + GENERATED_JSON + " file", e);
+            }
+        }
+        fileManifest.addFile(manifestFile);
+
+        Node generatedJson = buildManifestFile(settings, model, fileManifest, dependencies);
+        fileManifest.writeFile(manifestFile.toString(), Node.prettyPrintJson(generatedJson) + "\n");
+    }
+
+    private static Node buildManifestFile(
+            GoSettings settings,
+            Model model,
+            FileManifest fileManifest,
+            List<SymbolDependency> dependencies
+    ) {
+
+        List<SymbolDependency> nonStdLib = new ArrayList<>();
+        Optional<SymbolDependency> minStandard = Optional.empty();
+
+        for (SymbolDependency dependency : dependencies) {
+            if (!dependency.getDependencyType().equals(GoDependency.Type.STANDARD_LIBRARY.toString())) {
+                nonStdLib.add(dependency);
+            } else {
+                if (minStandard.isPresent()) {
+                    if (minStandard.get().getVersion().compareTo(dependency.getVersion()) < 0) {
+                        minStandard = Optional.of(dependency);
+                    }
+                } else {
+                    minStandard = Optional.of(dependency);
+                }
+            }
+        }
+
+        Map<StringNode, Node> manifestNodes = new HashMap<>();
+
+        Map<String, String> minimumDependencies = gatherMinimumDependencies(nonStdLib.stream());
+
+        Map<StringNode, Node> dependencyNodes = new HashMap<>();
+        for (Map.Entry<String, String> entry : minimumDependencies.entrySet()) {
+            dependencyNodes.put(StringNode.from(entry.getKey()),
+                    StringNode.from(entry.getValue()));
+        }
```

Comment:
will these be in stable order, or is a `TreeHash` needed for that?

Reply:
Yes these are stable, by means of line 128 `ObjectNode.objectNode(manifestNodes).withDeepSortedKeys();`